src/pkg/utils/helpers/files.go

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package helpers provides generic helper functions with no external imports
+package helpers
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mholt/archiver/v3"
+)
+
+// FindAndCopyFileFromArchive inspects an archive file, and copies the target file to the destination
+func FindAndCopyFileFromArchive(archivePath, targetFile, destinationDir string) error {
+	err := archiver.Walk(archivePath, func(f archiver.File) error {
+		if strings.HasSuffix(f.Name(), targetFile) {
```

Comment:
This won't work for directories - I believe it would need to be:

```suggestion
		if strings.HasPrefix(f.Name(), targetFile) {
```

Reply:
![image](https://github.com/defenseunicorns/zarf/assets/1096507/1b3610e5-95b4-4c7a-86b8-983ebc12f405)

I could be wrong but i think even in the case of a directory it must be suffix because the prefix is a `.`

You brought up a good point that this does not work with directories. I am going to re-work this to check if the file is a directory

Reply:
Nevermind, i found a much easier way to do this